src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/InterceptedProjectProperties.cs

```diff
@@ -27,10 +31,14 @@ public InterceptedProjectProperties(ImmutableArray<Lazy<IInterceptingPropertyVal
                 {
                     Requires.Argument(!string.IsNullOrEmpty(propertyName), nameof(valueProvider), "A null or empty property name was found");
 
-                    // CONSIDER: Allow duplicate intercepting property value providers for same property name.
-                    Requires.Argument(!builder.ContainsKey(propertyName), nameof(valueProviders), "Duplicate property value providers for same property name");
-
-                    builder.Add(propertyName, valueProvider);
+                    if (!builder.ContainsKey(propertyName))
+                    {
+                        builder.Add(propertyName, (new List<Lazy<IInterceptingPropertyValueProvider, IInterceptingPropertyValueProviderMetadata>> { valueProvider }, false));
+                    }
+                    else
+                    {
+                        builder[propertyName].Providers.Add(valueProvider);
+                    }
```

Comment:
This does a double lookup.  How about:

```suggestion
                    if (!builder.TryGetValue(propertyName, out var entry))
                    {
                        entry = (new List<Lazy<IInterceptingPropertyValueProvider, IInterceptingPropertyValueProviderMetadata>> { valueProvider }, false);
                        builder.Add(propertyName, entry);
                    }

                    entry.Providers.Add(valueProvider);
```